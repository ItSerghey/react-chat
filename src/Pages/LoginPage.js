import React from 'react';
import { withStyles } from 'material-ui';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography'
import Grid from 'material-ui/Grid';
import Paper from 'material-ui/Paper';
import Tabs, { Tab } from 'material-ui/Tabs';
import LoginForm from '../components/forms/LoginForm';
import SignupForm from '../components/forms/SignupForm';
import { Redirect } from 'react-router-dom';
import ErrorMessage from '../components/ErrorMessage';

const styles = theme => ({
  paper: {
    marginTop: 64 + theme.spacing.unit * 3,
    width: 500,
  },
  tabContent: {
    padding: theme.spacing.unit * 3,
  }
})

class LoginPage extends React.Component {

  state = {
    activeTab: 0,
  }

  handleChange = (event, value) => {
    this.setState({ activeTab: value });
  };

  render() {
    const { classes, signup, login, isAuthenticated,error } = this.props;
    const { activeTab } = this.state;

    if (isAuthenticated) {
      return (<Redirect to="/chat" />);
    }
    return (
      <React.Fragment>
        <AppBar>
          <Toolbar>
            <Typography variant="title" color="inherit" style={{ flex: 1 }}>
             YourCHAT
            </Typography>
          </Toolbar>
        </AppBar>

        <Grid container justify="center">
          <Grid item>
            <Paper className={classes.paper}>
              <AppBar position="static" color="default">
                <Tabs value={activeTab}
                  onChange={this.handleChange}
                  fullWidth
                >
                  <Tab label="Login" />
                  <Tab label="Sign Up" />
                </Tabs>
              </AppBar>
              <div className={classes.tabContent}>
                {activeTab === 0 && <LoginForm onSubmit={login} />}
                {activeTab === 1 && <SignupForm onSubmit={signup} />}
              </div>
            </Paper>
          </Grid>
        </Grid>
        <ErrorMessage error={error} />
      </React.Fragment>

    )
  }
}

export default withStyles(styles)(LoginPage);
