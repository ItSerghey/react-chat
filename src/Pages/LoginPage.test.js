/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import LoginPage from './LoginPage';

jest.mock('../components/forms/LoginForm', () => () => 'LoginForm');
jest.mock('../components/forms/SignupForm', () => () => 'SignupForm');
jest.mock('../components/ErrorMessage', () => () => 'ErrorMessage');

const mockProps = {
  signup: jest.fn(),
  login: jest.fn(),
  recieveAuth: jest.fn(),
  isAuthenticated: false,
  error: null,
};

describe('<LoginPage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter initialEntries={['/']}>
        <Route path="/(welcome)?" render={props => <LoginPage {...mockProps} {...props} />} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer // eslint-disable-next-line
      .create(
        <MemoryRouter initialEntries={['/']}>
          <Route path="/(welcome)?" render={props => <LoginPage {...mockProps} {...props} />} />
        </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
