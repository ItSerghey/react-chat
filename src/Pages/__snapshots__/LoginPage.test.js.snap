// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-198 MuiPaper-elevation4-204 MuiAppBar-root-190 MuiAppBar-positionFixed-191 MuiAppBar-colorPrimary-196 mui-fixed"
>
    <div
        className="MuiToolbar-root-225 MuiToolbar-gutters-226"
    >
        <h2
            className="MuiTypography-root-227 MuiTypography-title-233 MuiTypography-colorInherit-246"
            style={
                Object {
                    "flex": 1,
                  }
            }
        >
            YourCHAT
        </h2>
    </div>
</header>,
  <div
    className="MuiGrid-container-251 MuiGrid-justify-xs-center-268"
>
    <div
        className="MuiGrid-item-252"
    >
        <div
            className="MuiPaper-root-198 MuiPaper-elevation2-202 MuiPaper-rounded-199 LoginPage-paper-188"
        >
            <header
                className="MuiPaper-root-198 MuiPaper-elevation4-204 MuiAppBar-root-190 MuiAppBar-positionStatic-194 MuiAppBar-colorDefault-195"
            >
                <div
                    className="MuiTabs-root-341"
                >
                    <div
                        className="MuiTabs-flexContainer-342"
                    >
                        <div
                            className="MuiTabs-scroller-343 MuiTabs-fixed-344"
                            onScroll={[Function]}
                            role="tablist"
                            style={
                                Object {
                                    "marginBottom": 0,
                                  }
                            }
                        >
                            <div
                                className="MuiTabs-flexContainer-342"
                            >
                                <button
                                    aria-selected={true}
                                    className="MuiButtonBase-root-362 MuiTab-root-350 MuiTab-textColorInherit-352 MuiTab-selected-355 MuiTab-fullWidth-357"
                                    disabled={false}
                                    onBlur={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    onKeyUp={[Function]}
                                    onMouseDown={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseUp={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchMove={[Function]}
                                    onTouchStart={[Function]}
                                    role="tab"
                                    tabIndex="0"
                                    type="button"
                                >
                                    <span
                                        className="MuiTab-wrapper-358"
                                    >
                                        <span
                                            className="MuiTab-labelContainer-359"
                                        >
                                            <span
                                                className="MuiTab-label-360"
                                            >
                                                Login
                                            </span>
                                        </span>
                                    </span>
                                    <span
                                        className="MuiTouchRipple-root-368"
                                    />
                                </button>
                                <button
                                    aria-selected={false}
                                    className="MuiButtonBase-root-362 MuiTab-root-350 MuiTab-textColorInherit-352 MuiTab-fullWidth-357"
                                    disabled={false}
                                    onBlur={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    onKeyUp={[Function]}
                                    onMouseDown={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseUp={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchMove={[Function]}
                                    onTouchStart={[Function]}
                                    role="tab"
                                    tabIndex="0"
                                    type="button"
                                >
                                    <span
                                        className="MuiTab-wrapper-358"
                                    >
                                        <span
                                            className="MuiTab-labelContainer-359"
                                        >
                                            <span
                                                className="MuiTab-label-360"
                                            >
                                                Sign Up
                                            </span>
                                        </span>
                                    </span>
                                    <span
                                        className="MuiTouchRipple-root-368"
                                    />
                                </button>
                            </div>
                            <span
                                className="TabIndicator-root-365 TabIndicator-colorSecondary-367 MuiTabs-indicator-349"
                                style={
                                    Object {
                                        "left": 0,
                                        "width": 0,
                                      }
                                }
                            />
                        </div>
                    </div>
                </div>
            </header>
            <div
                className="LoginPage-tabContent-189"
            >
                LoginForm
            </div>
        </div>
    </div>
</div>,
  "ErrorMessage",
]
`;
